crates/client/rpc/src/lib.rs

```diff
@@ -937,3 +962,47 @@ where
         }
     }
 }
+
+/// Helper function to get filter events using address and keys
+///
+/// # Arguments
+///
+/// * `events` - A vector of all events
+/// * `address` - Address to use to filter the events
+/// * `keys` - Keys to use to filter the events. An event is filtered if any key is present
+/// * `max_results` - Optional, indicated the max events that need to be filtered
+///
+/// # Returns
+///
+/// * `(block_events: Vec<EventWrapper>, continuation_token: usize)` - A tuple of the filtered
+///   events and the first index which still hasn't been processed block_id and an instance of Block
+pub fn filter_events_by_params(
+    events: Skip<IntoIter<EventWrapper>>,
+    address: Option<Felt252Wrapper>,
+    keys: Vec<Vec<FieldElement>>,
+    max_results: Option<usize>,
+) -> (Vec<EventWrapper>, usize) {
+    let mut filtered_events = vec![];
+    let mut index = 0;
+
+    // Iterate on block events.
+    for event in events {
+        index += 1;
+        let match_from_address = address.map_or(true, |add| add == event.from_address);
```

Comment:
`add` is a bad abreviation for address. It sound like addition
addr is better

Reply:
fixed